### DISCUSSION TITLE
Create draft user with custom endpoint

### DISCUSSION THREAD

**Discussion by Silbad** [directus/directus#6611] [Archive]
Hello,

I would like to create a user (status: draft) from a custom endpoint. The big constraint is that I am not authenticated.
I don't want to use the invitation system because I want to record the first name, the last name and also the password that the user will have entered. I send a custom invitation email with a hook.
What is the best and most secure way to get permission to do this (accountability)?

```js
    router.post('/create', (req, res, next) => {
        const usersService = new ItemsService('directus_users', {
            schema: req.schema,
            accountability: req.accountability
        });

        const { first_name, last_name, email, password } = req.body;

        usersService.createOne({
            status: 'draft',
            first_name: first_name,
            last_name: last_name,
            email: email,
            password: password,
            role: 'xxxxxxxx-xxxxx-xxxx-xxxx-xxxxxxxxx'
        })
            .then((user) => {
                console.log(user)
            })
            .catch((error) => {
                console.log(error.message);
            });

        res.send('ok !')
    });
```

Thanks

**Comment by rijkvanzanten** [comment]
If you omit the accountability info when creating the service, eg here:

```js
const usersService = new ItemsService('directus_users', {
            schema: req.schema,
            accountability: req.accountability // <<<<<<<<<<<<<<<<
});
```

it'll allow you to do everything 👍🏻 . Make sure you handle any and all checks and constraints in your custom endpoint though!

**Reply by Silbad** [reply]
Ooh I don't try this ! Perfect ! Thanks !
